include/extractor/guidance/intersection.hpp

```diff
@@ -125,6 +127,12 @@ template <typename Self> struct EnableIntersectionOps
         auto comp = makeCompareAngularDeviation(angle);
         return std::min_element(self()->begin(), self()->end(), comp);
     }
+    // returns a non-const_interator
+    auto findClosestTurn(double angle)
+    {
+        auto comp = makeCompareAngularDeviation(angle);
+        return std::min_element(self()->begin(), self()->end(), comp);
+    }
```

Comment:
Non-const overload for the function above?

There is a trick involving a const-cast so you don't have to duplicate code.
Search around for how to implement const / non-const overloads.

Not sure if we need it here ..